core/src/main/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoEntityProvider.scala

```diff
@@ -76,25 +78,20 @@ class DataRepoEntityProvider(snapshotModel: SnapshotModel, requestArguments: Ent
     // they should be safe, but we should have layers of protection.
     val query = s"SELECT * FROM `${validateSql(dataProject)}.${validateSql(viewName)}.${validateSql(entityType)}` WHERE $pk = @pkvalue;"
     // generate query config, with named param for primary key
-    val queryConfig = QueryJobConfiguration.newBuilder(query)
+    val queryConfigBuilder = QueryJobConfiguration.newBuilder(query)
+      .setMaximumBytesBilled(config.bigQueryMaximumBytesBilled)
       .addNamedParameter("pkvalue", QueryParameterValue.string(entityName))
-      .build
-
-    // get pet service account key for this user
-    samDAO.getPetServiceAccountKeyForUser(googleProject, requestArguments.userInfo.userEmail) map { petKey =>
-
-      // get a BQ service (i.e. dao) instance, and use it to execute the query against BQ
-      val queryResource: Resource[IO, TableResult] = for {
-        bqService <- bqServiceFactory.getServiceForPet(petKey)
-        queryResults <- Resource.liftF(bqService.query(queryConfig))
-      } yield queryResults
```

Comment:
this has been restructured to be in terms of `IO` and not a nested map under `samDAO.getPetServiceAccountKeyForUser`